crates/core/c8y_api/src/http_proxy.rs

```diff
@@ -101,12 +101,19 @@ impl C8yEndPoint {
         // * <tenant_id>.<domain> eg: t12345.c8y.io
         // These URLs may be both equivalent and point to the same tenant.
         // We are going to remove that and only check if the domain is the same.
-        let tenant_uri = &self.c8y_host;
+        let (tenant_host, _port) = self
+            .c8y_host
+            .split_once(':')
+            .unwrap_or((&self.c8y_host, ""));
         let url = Url::parse(url).ok()?;
         let url_host = url.domain()?;
 
         let (_, host) = url_host.split_once('.').unwrap_or((url_host, ""));
-        let (_, c8y_host) = tenant_uri.split_once('.').unwrap();
+        let (_, c8y_host) = tenant_host.split_once('.').unwrap();
```

Comment:
Not sure this `unwrap` is safe. What's about a C8Y Edge tenant known only by its host name?

Reply:
Yeah, this is something that @reubenmiller and I were discussing in the morning and the conclusion was that this whole subdomain based check is fragile and must be replaced with a proper check that retrieves the real tenant URL behind any custom domain and validates the incoming URL against that. But, since that work is a bit more involved, we decided to create a separate issue for that, and just resolve the port aspect in this PR (keeping the existing fragile mechanism as-is for).

Reply:
So, let's go with this.

Reply:
Here is ticket for the custom domain issue: https://github.com/thin-edge/thin-edge.io/issues/2804

In the mean time, I've just updated the unwrap logic to at least avoid a panic.